src/utils/CustomPropTypes.js

```diff
@@ -57,4 +66,18 @@ function createMountableChecker() {
   return createChainableTypeChecker(validate);
 }
 
+function createKeyOfChecker(obj) {
+  function validate(props, propName, componentName) {
+    let propValue = props[propName];
+    if (!obj.hasOwnProperty(propValue)) {
+      let valuesString = JSON.stringify(Object.keys(obj));
+      return new Error(
+        ('Invalid prop `' + propName + '` of value `' + propValue + '` ') +
+        ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.')
```

Comment:
We're using ES6 you can use string interpolation here to clean this up: http://babeljs.io/docs/learn-es6/#template-strings